### ISSUE TITLE
`useIntersection`: unobserve after triggering `appear` callback

### ISSUE THREAD

**Issue by Eldzej** [stimulus-use/stimulus-use#370]
Is it possible to somehow call the `appear` event added by `useIntersection` only once? Or manually call `unobserve` on the created `IntersectionObserver`?

In my usecase, I am initializing a library when element enters the viewport, but with the default options, the code gets invoked every time when element enters the viewport, which is rather impractical.

For the moment, I am managing it by checking a custom property on the controller, but wondering if there isn't a "cleaner" way

```
connect() {
        this.isLibraryLoaded = false
        useIntersection(this)
    }

    appear() {
        if (!this.isLibraryLoaded) {
            this.isLibraryLoaded = true
            this.loadLibrary()
        }
    }

    loadLibrary() {
        /* do stuff */
    }
```

Sorry if this is stupid question or answered somewhere, I could not find it...

**Comment by marcoroth** [comment]
Hey @Eldzej, currently there is no built-in option to just call it once. But you can call `unobserve` manually by obtaining a reference to the function from the `useIntersection()` function call. It returns an array with two functions, the first one is `observe` and the second one `unobserve`. So you could do something like:
```js
export default class extends Controller { 
  connect() {
    const [_observe, unobserve] = useIntersection(this)
    this.unobserve = unobserve
  }
  
  appear(entry) {  
    console.log("appear", entry)
    this.unobserve()
  }
}
```
Here's a working example: https://codepen.io/marcoroth/pen/YzRrrRr?editors=1010

But the brings up the question. I feel like we could also expose the actual `IntersectionObserver` instance by either assigning it to a property on the controller or by passing in the reference to the `appear` and `disappear` callback as the second argument.

So that you could do something like:
```js
export default class extends Controller { 
  connect() {
    useIntersection(this)
  }
  
  appear(entry, observer) {  
    console.log("appear", entry)
    observer.unobserve(entry.target)
  }
}
```

I hope this helps

**Comment by Eldzej** [comment]
Yeah, that is something I have been trying, thanks.

I have been trying to get the reference incorrectly (looking at code, I thought it should be object, rather than array), and I kept getting undefined as a result

I think exposing the observer instance as part of the controller (similar to how the helper functions are) itself might be good? That is how I have been using IntersectionObserver before stumbling upon this repo.

```
this.observer = new IntersectionObserver(entries => {
    if (entries[0].isIntersecting && this.inView !== true) {
        this.inView = entries[0].isIntersecting
        this.loadLibrary()
        this.observer.disconnect()
    }
}, options)
```

**Comment by marcoroth** [comment]
I just opened https://github.com/stimulus-use/stimulus-use/pull/371 to allow to access the `observer` itself in the callbacks.

**Comment by marcoroth** [comment]
Resolved via #371
